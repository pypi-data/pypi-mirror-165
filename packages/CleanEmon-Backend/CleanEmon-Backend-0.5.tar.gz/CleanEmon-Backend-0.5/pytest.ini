[pytest]
markers =
    projectwise: Mark a test as project-dependant
    slow: Mark a test which is expected to be slow on execution
filterwarnings =
    ignore::DeprecationWarning
    ignore::RuntimeWarning
    ignore::UserWarning:lab