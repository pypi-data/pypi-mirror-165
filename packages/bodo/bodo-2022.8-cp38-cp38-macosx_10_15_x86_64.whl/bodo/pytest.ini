[pytest]
markers =
    slow: marks tests as slow (deselect with '-m "not slow"')
    smoke: marks tests as basic smoke tests to run on every PR (select with '-m "smoke"')
    s3: s3 related tests
    hdfs: hdfs related tests
    firsthalf: first half of the test suite
    firstthird: first third of the test suite
    secondthird: second third of the test suite
    lastthird: last third of the test suite
    single_mod: single python module from which to run tests
    iceberg: iceberg related tests

addopts = -p no:faulthandler
