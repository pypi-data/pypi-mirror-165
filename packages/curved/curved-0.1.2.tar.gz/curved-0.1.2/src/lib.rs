use ndarray::{s, Axis, Array1, ArrayViewMut1, ArrayView1, ArrayView2, CowArray, Ix1, Ix2, Slice};
use numpy::{convert::IntoPyArray, PyArray1, PyArray2};
use pyo3::prelude::{pymodule, pyfunction, wrap_pyfunction, Py, PyModule, PyResult, Python};


pub fn rdp(points: ArrayView2<'_, f64>, epsilon: f64) -> Array1<bool> {
    // Generate a mask boolean array, which will be the result.
    let mut mask = Array1::from_elem((points.len_of(Axis(0)),), false);
    mask[0] = true;
    let mask_len = mask.len();
    mask[mask_len - 1] = true;

    // Run the recursive RDP algorithm
    rdp_recurse(points, None, mask.view_mut(), epsilon.powi(2));

    mask
}


struct LineStartPointBuffer<'a> {
    vectors: CowArray<'a, f64, Ix2>,
    magnitudes_2: CowArray<'a, f64, Ix1>,
}


impl LineStartPointBuffer<'_> {
    fn from_points<'a>(points: ArrayView2<f64>) -> LineStartPointBuffer<'a> {
        let start = points.slice(s![0, ..]);
        let vectors = &points - &start;
        let magnitudes_2 = (&vectors * &vectors).sum_axis(Axis(1));
        LineStartPointBuffer { vectors: vectors.into(), magnitudes_2: magnitudes_2.into() }
    }

    fn subset(&self, slice: Slice) -> LineStartPointBuffer {
        LineStartPointBuffer {vectors: self.vectors.slice(s![slice, ..]).into(),
            magnitudes_2: self.magnitudes_2.slice(s![slice]).into()}
    }
}


fn line_point_distances_2(
    start: ArrayView1<'_, f64>,
    end: ArrayView1<'_, f64>,
    buffer: &LineStartPointBuffer<'_>
) -> Array1<f64> {
    // Calculate the unit vector from the start (A) to the end (B)
    let ab: Array1<f64> = &end - &start;
    let ab_magnitude = ab.dot(&ab).sqrt();
    let ab_unit = ab / ab_magnitude;

    // Project the point (C) to start (A) vector onto the unit vector using a dot product.
    // It should return the magnitude of the vector from the start (A) to the point closest
    // to the AB line (D).
    let ad_magnitudes_2 = buffer.vectors.dot(&ab_unit).mapv(|v| v.powi(2));
    &buffer.magnitudes_2 - &ad_magnitudes_2
}

fn rdp_recurse(points: ArrayView2<'_, f64>, buffer: Option<LineStartPointBuffer<'_>>, mut mask: ArrayViewMut1<'_, bool>, epsilon_2: f64) {
    // Get the start and end points of the curve
    let start = points.slice(s![0, ..]);
    let end = points.slice(s![-1, ..]);

    // Calculate some buffered line start to point values.
    let buffer = buffer.unwrap_or_else(|| LineStartPointBuffer::from_points(points));

    // Calculate the distance squared between the line and each point.
    let distances_2 = line_point_distances_2(start, end, &buffer);

    // Find the point with the maximum distance from the line joining the endpoints.
    let mut d_2_max = 0.0;
    let mut i_max: usize = 0;
    for (i, d) in distances_2.iter().enumerate().take(distances_2.len() - 1).skip(1) {
        if d > &d_2_max {
            i_max = i;
            d_2_max = *d;
        }
    }

    // If that point is further away from the line joining the endpoints than epsilon^2,
    // mark it as retained and recurse the algorithm for the line joining start->point
    // and point->end.
    if d_2_max > epsilon_2 {
        mask[i_max] = true;
        rdp_recurse(points.slice(s![..=i_max, ..]),
            Some(buffer.subset(Slice::from(..=i_max))),
            mask.slice_mut(s![..=i_max]),
            epsilon_2);
        rdp_recurse(points.slice(s![i_max.., ..]), None, mask.slice_mut(s![i_max..]), epsilon_2);
    }
}

#[pymodule]
fn _rustlib(_py: Python<'_>, m: &PyModule) -> PyResult<()> {
    #[pyfunction]
    fn rdp(
        py: Python<'_>,
        points: &PyArray2<f64>,
        epsilon: f64
    ) -> Py<PyArray1<bool>> {
        let points = points.readonly();
        crate::rdp(points.as_array(), epsilon).into_pyarray(py).to_owned()
    }

    m.add_function(wrap_pyfunction!(rdp, m)?)?;

    Ok(())
}
