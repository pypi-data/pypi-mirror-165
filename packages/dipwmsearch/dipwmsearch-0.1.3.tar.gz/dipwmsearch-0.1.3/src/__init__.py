# SOL https://py-pkgs.org/04-package-structure.html
# read version from installed package
from importlib.metadata import version
__version__ = version("dipwmsearch")

# ##################################################
print ( __version__ )
# ##################################################

from .AhoCorasick import *
# from .BestWords import *
from .Block import *
from .diPwm import *
from .Enumerate import *
from .SemiNaive import *
# from .Super import *

#
