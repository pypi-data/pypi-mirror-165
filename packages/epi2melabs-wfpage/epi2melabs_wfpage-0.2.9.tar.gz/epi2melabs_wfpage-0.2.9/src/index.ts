import {
  JupyterFrontEnd,
  JupyterFrontEndPlugin
} from '@jupyterlab/application';

import { ISettingRegistry } from '@jupyterlab/settingregistry';

import { IDocumentManager } from '@jupyterlab/docmanager';

import { MainAreaWidget } from '@jupyterlab/apputils';

import { ILauncher } from '@jupyterlab/launcher';

import { labsLogoIcon } from './asset';

import { Launcher } from './widget';

const PLUGIN_ID = '@epi2melabs/epi2melabs-wfpage:plugin';

const COMMAND = 'create-epi2me-labs-launcher';

const CATEGORY = 'EPI2ME Labs';

/**
 * Initialization data for the epi2melabs-wfpage extension.
 */
const plugin: JupyterFrontEndPlugin<void> = {
  id: PLUGIN_ID,
  autoStart: true,
  requires: [ILauncher, ISettingRegistry, IDocumentManager],
  activate: (
    app: JupyterFrontEnd,
    launcher: ILauncher,
    settings: ISettingRegistry,
    docTrack: IDocumentManager
  ) => {
    const { commands, shell } = app;

    const createLauncherWidget = (
      shell: JupyterFrontEnd.IShell,
      docTrack: IDocumentManager,
      plugin_settings: ISettingRegistry.ISettings
    ) => {
      const content = new Launcher(app, docTrack, plugin_settings);
      const widget = new MainAreaWidget<Launcher>({ content });
      widget.title.label = 'EPI2ME Labs';
      shell.add(widget, 'main');
    };

    Promise.all([app.restored, settings.load(PLUGIN_ID)]).then(
      ([, _settings]) => {
        commands.addCommand(COMMAND, {
          caption: 'Create an EPI2ME Labs launcher',
          label: 'EPI2ME Labs',
          icon: labsLogoIcon,
          execute: () => createLauncherWidget(shell, docTrack, _settings)
        });

        createLauncherWidget(shell, docTrack, _settings);

        if (launcher) {
          launcher.add({
            command: COMMAND,
            category: CATEGORY
          });
        }

        app.commands.execute('filebrowser:hide-main');
      }
    );
  }
};

export default plugin;
