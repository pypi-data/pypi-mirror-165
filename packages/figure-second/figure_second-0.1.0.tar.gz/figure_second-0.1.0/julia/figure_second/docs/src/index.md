# Welcome

`figure_second` is a layout first approach to plotting based on the ideas of the python
library [figurefirst](https://flyranch.github.io/figurefirst/). The general workflow of the library
is to define a layout of graphs in inkscape, label each object with an XML `id`, and then plot 
*into* these objects using common julia plotting libraries (`Makie.jl`, `Plots.jl`, etc).

```@contents
Depth = 2
```

## Example

TODO

## API Reference

For figure layout, the most useful functions will be [`updater`](@ref) to construct a common object,
[`plot_figures`](@ref) to place figures in the document, [`relative_dimensions`](@ref) for sizing 
plots correctly. 

The [`ids`](@ref) function can sometimes be useful for debugging purposes.

```@autodocs
Modules = [figure_second, figure_second.python_bindings]
```
