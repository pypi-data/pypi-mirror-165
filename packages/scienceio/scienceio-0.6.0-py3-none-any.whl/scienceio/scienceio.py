import configparser
import json
import os
import requests
import types

from enum import Enum
from pathlib import Path
from typing import Optional, Union

CONFIG = types.SimpleNamespace(
    API_URL="https://api.aws.science.io/v1",
    HELP_EMAIL="api_support@science.io",
    SETTINGS_DIR=Path.home() / ".scienceio",
    SETTINGS_FILE_PATH=Path.home() / ".scienceio" / "config",
)


class ScienceIOError(Exception):
    """Base class for all exceptions that are raised by the ScienceIO SDK."""


class TimeoutError(ScienceIOError):
    """Raised when a call to the ScienceIO API times out."""

    def __init__(self, msg):
        self.msg = msg


class ResponseFormat(str, Enum):
    JSON = "application/json"

    def format_response(self, response):
        """
        Defines response format

        Args:
            response: string with desired format
        """
        if self is ResponseFormat.JSON:
            return response.json()
        else:
            raise NotImplementedError("Unknown response format")

    def create_headers(self, api_id: str, api_secret: str) -> dict:
        """
        Create headers for API call

        Args:
            api_id:  API Key, obtained from api.science.io
            api_secret:  API secret, obtained from api.science.io

        Returns:
            Dict containing content type and API credentials
        """
        return {
            "Content-Type": self.value,
            "x-api-id": api_id,
            "x-api-secret": api_secret,
        }


class ScienceIO(object):
    def __init__(
        self,
        response_format: ResponseFormat = ResponseFormat.JSON,
        timeout: Optional[Union[int, float]] = 1200,
    ):
        """Initializer for the ScienceIO client. The client will attempt to
        configure itself by trying to read from environment variables, if set.
        If unable to, the config values will be read from the ScienceIO config
        file.

        Args:
            response_format:
                Format to use for responses (default: `ResponseFormat.JSON`).
            timeout:
                Amount of time to wait before timing out network calls, in seconds.
                If `None`, timeouts will be disabled. (default: 1200)
        """

        self.response_format = response_format
        self.timeout = timeout

        # Create a persistent session across requests.
        # https://docs.python-requests.org/en/master/user/advanced/
        self.session = requests.Session()
        self.session.params = {}

        # Lazy loading of configuration (no need to try and load the settings if user specifies
        # their own API ID and secret). Also, this prevents breakage when in envs with no
        # settings file, such as test environments or hosted Jupyter notebooks.
        config = None

        def get_config_value(key: str) -> str:
            nonlocal config
            if config is None:
                config = configparser.RawConfigParser()
                config.read(CONFIG.SETTINGS_FILE_PATH)

            return config["SETTINGS"][key]

        # Handles config values from user arguments, config file, and user
        # input, in that order from most to least preferred.
        write_out_conf = False

        def get_value(initial: Optional[str], key: str, human_name: str) -> str:
            nonlocal write_out_conf
            if initial is None:
                try:
                    return get_config_value(key)
                except KeyError:
                    user_input = str(
                        input(f"Please provide your ScienceIO API key {human_name}: ")
                    )

                    # User input was collected, flag the config file for rewriting.
                    write_out_conf = True
                    return user_input

            return initial

        # API endpoints to use.
        self.api_url = os.environ.get("SCIENCEIO_API_URL", CONFIG.API_URL)
        self.annotate_url = f"{self.api_url}/annotate"

        # API key and secret (with extra handling for env vars, config file, and user prompts).
        self.api_id = get_value(os.environ.get("SCIENCEIO_KEY_ID"), "KEY_ID", "id")
        self.api_secret = get_value(
            os.environ.get("SCIENCEIO_KEY_SECRET"), "KEY_SECRET", "secret"
        )

        # Write out the API key ID and secret if either or both of those values
        # needed user input.
        if write_out_conf:
            # Create a new `ConfigParser` to hold the configuration we want to
            # write to the config file.
            new_conf = configparser.ConfigParser()
            new_conf["SETTINGS"] = {
                "KEY_ID": self.api_id,
                "KEY_SECRET": self.api_secret,
            }

            # Create the config directory (and any parents) if it does not
            # already exist.
            CONFIG.SETTINGS_DIR.mkdir(parents=True, exist_ok=True)

            # Write out the new config.
            with open(CONFIG.SETTINGS_FILE_PATH, "w") as fp:
                new_conf.write(fp)

    def annotate(self, text: str) -> dict:
        """Annotates a block of text using the ScienceIO API.

        Args:
            text (str): The text to annotate.

        Raises:
            ValueError: Raised if the input text is not a `str`.
            TimeoutError: Raised if the annotation request exceeds the timeout
                limit.

        Returns:
            dict: The annotation data, as a Python dictionary.
        """
        if not isinstance(text, str):
            raise ValueError("annotate argument must be a string.")

        payload = json.dumps(
            {
                "text": text,
            }
        )

        headers = self.response_format.create_headers(
            api_id=self.api_id,
            api_secret=self.api_secret,
        )

        try:
            response = self.session.post(
                self.annotate_url,
                data=payload,
                headers=headers,
                timeout=self.timeout,
            )
        except requests.Timeout:
            raise TimeoutError("annotation timed out, please try again") from None

        result = self.response_format.format_response(response)

        return result
