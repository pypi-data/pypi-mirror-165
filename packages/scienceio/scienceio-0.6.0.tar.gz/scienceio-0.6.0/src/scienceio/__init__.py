import os

from .scienceio import ScienceIO  # noqa

__version__ = os.environ.get("SCIENCEIO_SDK_VERSION", "v0.0.1")
