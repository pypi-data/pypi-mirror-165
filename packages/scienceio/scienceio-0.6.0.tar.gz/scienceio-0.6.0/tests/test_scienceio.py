import pytest

from src.scienceio import ScienceIO


@pytest.fixture(scope="session", autouse=True)
def scio():
    """
    Init ScienceIO client, using .scio/config
    :return: our client
    """
    yield ScienceIO()


def test_input_is_string(scio):
    """
    Test providing something other than a string returns an error
    """
    query_text = {
        "text": "The COVID-19 pandemic has shown a markedly low proportion of cases among children 1-4. Age disparities in observed cases could be  explained by children having lower susceptibility to infection, lower  propensity to show clinical symptoms or both."
    }
    with pytest.raises(ValueError, match="annotate argument must be a string."):
        scio.annotate(text=query_text)
        raise ValueError("annotate argument must be a string.")


def test_annotate(scio, snapshot):
    """
    Test annotate request with a small text snippet
    """
    query_text = (
        "The COVID-19 pandemic has shown a markedly low proportion of "
        "cases among children 1-4. Age disparities in observed cases could be "
        "explained by children having lower susceptibility to infection, lower "
        "propensity to show clinical symptoms or both."
    )

    response = scio.annotate(text=query_text)

    # 1. Test response is a dict
    assert isinstance(response, dict)

    # 2. Test we get text back
    assert response.get("text") == query_text

    # 3. Test character length match
    assert response.get("characters") == 245

    # 4. Are we billing 1 unit
    assert response.get("units_number") == 1

    # 5. Do we get a request_id
    assert "request_id" in response
    del response["request_id"]

    # 6. General snapshot
    assert response == snapshot
