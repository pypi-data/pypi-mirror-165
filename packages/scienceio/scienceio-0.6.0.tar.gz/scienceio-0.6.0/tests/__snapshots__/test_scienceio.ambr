# name: test_annotate
  dict({
    'annotations': list([
      dict({
        'concept_id': 'UMLS:C5203670',
        'concept_name': 'COVID-19',
        'concept_type': 'Medical Conditions',
        'pos_end': 12,
        'pos_start': 4,
        'score_id': 0.9997,
        'score_type': 0.9999,
        'text': 'COVID-19',
      }),
      dict({
        'concept_id': 'UMLS:C1615608',
        'concept_name': 'Pandemics',
        'concept_type': 'Anatomy & Physiology',
        'pos_end': 21,
        'pos_start': 13,
        'score_id': 0.9998,
        'score_type': 0.9993,
        'text': 'pandemic',
      }),
      dict({
        'concept_id': 'UMLS:C0008059',
        'concept_name': 'Child',
        'concept_type': 'Context',
        'pos_end': 81,
        'pos_start': 73,
        'score_id': 0.8162,
        'score_type': 0.8118,
        'text': 'children',
      }),
      dict({
        'concept_id': 'UMLS:C0001779',
        'concept_name': 'Age',
        'concept_type': 'Anatomy & Physiology',
        'pos_end': 90,
        'pos_start': 87,
        'score_id': 0.9999,
        'score_type': 0.9999,
        'text': 'Age',
      }),
      dict({
        'concept_id': 'UMLS:C0008059',
        'concept_name': 'Child',
        'concept_type': 'Context',
        'pos_end': 151,
        'pos_start': 143,
        'score_id': 0.8046,
        'score_type': 0.7968,
        'text': 'children',
      }),
      dict({
        'concept_id': 'UMLS:C0009450',
        'concept_name': 'Communicable Diseases',
        'concept_type': 'Medical Conditions',
        'pos_end': 192,
        'pos_start': 183,
        'score_id': 0.6927,
        'score_type': 0.9999,
        'text': 'infection',
      }),
    ]),
    'characters': 245,
    'text': 'The COVID-19 pandemic has shown a markedly low proportion of cases among children 1-4. Age disparities in observed cases could be explained by children having lower susceptibility to infection, lower propensity to show clinical symptoms or both.',
    'units_number': 1,
  })
# ---
